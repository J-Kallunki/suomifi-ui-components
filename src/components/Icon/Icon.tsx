import React, { Component } from 'react';
import styled from '@emotion/styled';
import { Svg, SvgProps } from '../Svg/Svg';

export interface IconProps extends SvgProps {}

const StyledSvg = styled((props: SvgProps) => <Svg {...props} />)`
  display: inline-block;
  width: 24px;
  height: 24px;
`;

export class Icon extends Component<IconProps> {
  static defaultProps = {
    labelName: 'icon',
  };

  render() {
    return <StyledSvg {...this.props} />;
  }
}
