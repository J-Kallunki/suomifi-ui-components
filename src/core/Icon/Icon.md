```js
<Icon ariaLabel="Login here" className="my-icon--test" testId="icon" />
```
