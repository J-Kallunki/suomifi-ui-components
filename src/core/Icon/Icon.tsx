import React, { Component } from 'react';
import { defaultPropsTheme, objValue } from '../utils';
import classnames from 'classnames';
import {
  default as CompIcon,
  IIconProps as ICompIconProps,
} from '../../components/Icon/Icon';
import { icons, IconKeys } from '../theme/icons';
import iconLoginSvg from '../theme/icons/icon-login.svg';
import { Omit } from '../utils/typescript';

export interface IIconProps extends Omit<ICompIconProps, 'src'> {
  icon?: IconKeys;
  src?: string;
}

const className = 'fi-icon';

/**
 * General icon-component
 */
export default class Icon extends Component<IIconProps> {
  static defaultProps = defaultPropsTheme(CompIcon, theme => ({
    src: iconLoginSvg, // TODO
    color: theme.colors.brandColor,
  }));

  render() {
    const { src, icon, className: propsClassName, ...passProps } = this.props;
    const iconsIcon = (icon: IconKeys) => {
      const iconSrc = objValue(icons, icon);
      return !!iconSrc ? iconSrc : src;
    };
    const iconSrc = !!icon ? iconsIcon(icon) : src;
    const defaultIcon = iconLoginSvg;
    return (
      <CompIcon
        {...passProps}
        src={!!iconSrc ? iconSrc : defaultIcon}
        className={classnames(propsClassName, className)}
      />
    );
  }
}
