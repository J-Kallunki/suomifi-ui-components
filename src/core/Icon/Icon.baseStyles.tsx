import { css } from '@emotion/core';
import { SuomifiIconProps } from './Icon';

export const iconBaseStyles = ({
  mousePointer = false,
}: SuomifiIconProps) => css`
  display: inline-block;
  vertical-align: baseline;
  transform: translateY(0.1em);
  ${!!mousePointer &&
    `cursor: pointer;
      & * {
        cursor: inherit;
      }
  `}
`;
