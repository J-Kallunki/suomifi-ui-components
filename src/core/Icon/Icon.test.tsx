import React from 'react';
import { render } from 'react-testing-library';

import Icon from './Icon';

test('calling render with the same component on the same container does not remount', () => {
  const svgRenderer = render(<Icon />);
  const { getByTestId, rerender } = svgRenderer;
  expect(getByTestId('icon')).toBeDefined();

  // re-render the same component with different props
  rerender(<Icon />);
  expect(getByTestId('icon')).toBeDefined();
});
