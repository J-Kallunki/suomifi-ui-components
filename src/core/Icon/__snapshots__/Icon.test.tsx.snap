// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`calling render with the same component on the same container does not remount 1`] = `
.emotion-0 {
  display: inline-block;
  vertical-align: bottom;
}

<svg
  aria-hidden="true"
  class="emotion-0"
  data-testid="icon"
  focusable="false"
  height="1em"
  style="fill: #00357A;"
  viewBox="0 0 24 24"
  width="1em"
>
  <path
    d="M22.01 0c1.103 0 2 .897 2 2v20c0 1.103-.897 2-2 2H10c-1.103 0-2-.897-2-2v-5a1 1 0 0 1 2 0v5h12.01V2H10v5a1 1 0 0 1-2 0V2c0-1.103.897-2 2-2h12.01zm-9.717 15.293L14.586 13H1a1 1 0 0 1 0-2h13.586l-2.293-2.293a.999.999 0 1 1 1.414-1.414l4 4a.999.999 0 0 1 .216 1.089.999.999 0 0 1-.216.325l-4 4a.997.997 0 0 1-1.414 0 .999.999 0 0 1 0-1.414z"
  />
</svg>
`;
